source/s3_util.c

```diff
@@ -398,44 +417,57 @@ void aws_s3_add_user_agent_header(struct aws_allocator *allocator, struct aws_ht
     aws_byte_buf_clean_up(&user_agent_buffer);
 }
 
-int aws_s3_parse_content_range_response_header(
-    struct aws_allocator *allocator,
-    struct aws_http_headers *response_headers,
+int aws_s3_parse_content_range_cursor(
+    struct aws_byte_cursor content_range_cursor,
     uint64_t *out_range_start,
     uint64_t *out_range_end,
     uint64_t *out_object_size) {
-    AWS_PRECONDITION(allocator);
-    AWS_PRECONDITION(response_headers);
 
-    struct aws_byte_cursor content_range_header_value;
+    /* Expected Format of header is: "bytes StartByte-EndByte/TotalObjectSize" */
```

Comment:
does s3 always respect that or there are cases where it skips start and end (which is valid in http spec)